src/Disks/ObjectStorages/S3/registerDiskS3.cpp

```diff
@@ -122,15 +123,28 @@ void registerDiskS3(DiskFactory & factory)
         if (uri.key.back() != '/')
             throw Exception(ErrorCodes::BAD_ARGUMENTS, "S3 path must ends with '/', but '{}' doesn't.", uri.key);
 
-        auto [metadata_path, metadata_disk] = prepareForLocalMetadata(name, config, config_prefix, context);
-
-        auto metadata_storage = std::make_shared<MetadataStorageFromDisk>(metadata_disk, uri.key);
         S3Capabilities s3_capabilities = getCapabilitiesFromConfig(config, config_prefix);
+        std::shared_ptr<S3ObjectStorage> s3_storage;
+
+        MetadataStoragePtr metadata_storage;
+        if (name == "simple_s3")
```

Comment:
`simple_s3` is a bad name, because it means that other `s3` disk type is not simple. That would look strange. I think the main idea of your disk is that it doesn't use local metadata. So the name should represent that. Maybe something like `s3_external` (because `s3_without_local_metadata` seems too long)?

Reply:
s3_external actually does not underline that the metadata is in S3, but can be interpreted in the opposite way I would say.
I'm fine with s3_plain (at first I used it, but later changed to simple)

Reply:
ok, let's stop on `s3_plain`

Reply:
Will we have additional disk type for each object storage? (ie blob, gcs (in future))